src/main/js/Widgets/MultipleSelect.js

```diff
@@ -502,6 +502,11 @@ class MultipleSelect {
       }
       option.setSelected(true);
 
+      // Preserve the order of selected options.
+      const parent = option.domElement.parentNode;
+      option.removeFromDOM();
+      parent.appendChild(option.domElement);
```

Comment:
If I am reading this correctly, we'll move the option to end of the each time you select, so that we can preserve the order. 

The downside of doing this is that it will change the order in the DOM as well. So for the case of displaying something alphabetized such as locales, if you select a value, it will move the value in the DOM and then if you de-select it, it will now no longer be where you expect it in the list. 

This may be the simplest way to fix it. We could also disable the initial `select` control and build a new one with only the options we want, or we could append hidden fields and skip the `select`. We may want to experiment a bit more to see if we can solve it without modifying the order as they appear in the DOM since that may be important for the intended UX. 

We should be able to write a test for this as well.